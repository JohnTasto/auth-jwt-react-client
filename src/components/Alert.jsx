import React, { Component } from 'react'
import { connect } from 'react-redux'
import { actions as errorActions } from '../ducks/error'


export class Alert extends Component {
  render() {
    if (this.props.error) {
      return (
        <div className="alert alert-danger alert-dismissible" role="alert">
          <button type="button" className="close" aria-label="Close" onClick={this.props.clear}>
            <span aria-hidden="true">&times;</span>
          </button>
          <strong>Error:</strong> <span>{this.props.error}</span>
        </div>
      )
    }
    return null
  }
}

export default connect(
  state => ({
    error: state.error,
  }),
  errorActions,
  (...props) =>
    Object.assign({}, ...props),
)(Alert)
