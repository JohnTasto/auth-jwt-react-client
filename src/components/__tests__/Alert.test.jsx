/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Alert } from '../Alert'


describe('<Alert />', () => {
  test('Renders alert if there is an error', () => {
    const wrapper = shallow(<Alert error="ERROR!" />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  test('Renders nothing if there is no error', () => {
    const wrapper = shallow(<Alert />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  test('Calls action error.clear when close button is clicked', () => {
    const props = {
      error: 'ERROR!',
      clear: jest.fn(),
    }
    const wrapper = shallow(<Alert {...props} />)

    wrapper.find('button').simulate('click')

    expect(props.clear).toHaveBeenCalled()
  })
})
