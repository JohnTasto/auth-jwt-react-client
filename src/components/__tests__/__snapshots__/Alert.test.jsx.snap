// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Renders alert if there is an error 1`] = `
<div
  className="alert alert-danger alert-dismissible"
  role="alert"
>
  <button
    aria-label="Close"
    className="close"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
  <strong>
    Error:
  </strong>
   
  <span>
    ERROR!
  </span>
</div>
`;

exports[`<Alert /> Renders nothing if there is no error 1`] = `null`;
